crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -131,7 +147,20 @@ impl Settlement {
         persistence: &infra::Persistence,
         chain: &Chain,
     ) -> Result<Self, Error> {
-        let auction = persistence.get_auction(settled.auction_id).await?;
+        let (auction, solver_winning_solutions) = tokio::try_join!(
+            async {
+                persistence
+                    .get_auction(settled.auction_id)
+                    .await
+                    .map_err(Error::from)
+            },
+            async {
+                persistence
+                    .get_solver_winning_solutions(settled.auction_id, settled.solver)
+                    .await
+                    .map_err(Error::from)
+            }
```

Comment:
nit: could be made more concise with `futures::TryFutureExt`.
```suggestion
            persistence
                .get_auction(settled.auction_id)
                .map_err(Error::from),
            persistence
                .get_solver_winning_solutions(settled.auction_id, settled.solver)
                .map_err(Error::from),
```